package/rebuild-todo

```diff
@@ -234,8 +249,26 @@ for pkg in "${packages[@]}"; do
             fi
         fi
         if ! ((NO_PUBLISH)); then
-            pkgctl release --db-update $REPO -m "$message"
+            if pkgctl release --db-update $REPO -m "$message"; then
+                remove_from_rebuilt_packages_list "$pkg"
```

Comment:
Can't you drop the single element like described here?

Reply:
Unless I'm missing something, this method doesn't actually drop the element from the array but replace it by an empty/blank (yet existing) one.

To take the same example as from the link:

```
$ cat test.sh
#!/bin/bash

array=(test1)
delete=test1
array=( "${array[@]/$delete}" )

echo "Array element = ${array[@]}"
echo "Array appears empty but..."
echo "Number of array element = ${#array[@]}"

if ((${#array[@]})); then
        echo "There are items in array"
else
        echo "There are no item in array"
fi
```

```
$ ./test.sh
Array element =
Array appears empty but...
Number of array element = 1
There are items in array
```

This requires additional *sketchy* tinkering to format the list properly but, more importantly and as seen in the example above, it makes [the check that verifies if the array is empty or not](https://github.com/archlinux/contrib/pull/74/files#diff-c81afe64686ce8ada9f0a809f25a7098a3c8227a267eae6a5a306053b18691a0R266) failing because the array is not actually empty (it just contains empty/blank elements).

Unfortunately, it doesn't seem like bash has a simple and builtin mechanism to drop element from an array (e.g. `array-=element`, as opposed to `array+=element`) :disappointed:

Reply:
Actually I found a [comment](https://stackoverflow.com/questions/16860877/remove-an-element-from-a-bash-array#comment60100567_16861932) from the link that confirms the above: "Note: this may set the respective value to nothing, but the element will still be in the array."